tools/slang-capability-generator/capability-generator-main.cpp

```diff
@@ -126,32 +227,39 @@ struct CapabilityDefParser
                 m_sink->diagnose(nameToken.loc, Diagnostics::undefinedIdentifier, nameToken);
                 return SLANG_FAIL;
             }
-            if (!(advanceIf(TokenType::OpAnd) || advanceIf(TokenType::OpAdd)))
+            if (!(advanceIf(TokenType::OpAdd)))
                 break;
         }
         return SLANG_OK;
     }
 
     SlangResult parseExpr(CapabilityDisjunctionExpr& expr)
     {
+        AddCapabilityOptions addOption = AddCapabilityOptions::Union;
         for (;;)
         {
             CapabilityConjunctionExpr conjunction;
+            conjunction.sourceLoc = this->m_tokenReader.m_cursor->getLoc();
             SLANG_RETURN_ON_FAIL(parseConjunction(conjunction));
+            conjunction.howToJoinWithNextSetFlag = addOption;
+
             expr.conjunctions.add(conjunction);
+            addOption = AddCapabilityOptions::Union;
             if (!advanceIf(TokenType::OpBitOr))
                 break;
+            if (advanceIf(TokenType::OpBitAnd))
+                addOption = AddCapabilityOptions::InclusiveJoin;
         }
         return SLANG_OK;
     }
 
-    SlangResult parseDefs()
+    SlangResult parseDefs(CapabilitySharedContext& capabilitySharedContext)
     {
         auto tokens = m_lexer->lexAllSemanticTokens();
         m_tokenReader = TokenReader(tokens);
         for (;;)
         {
-            RefPtr<CapabilityDef> def = new CapabilityDef();
+            RefPtr<CapabilityDef> def = new CapabilityDef(capabilitySharedContext);
```

Comment:
Does this have to be allocated from heap memory?
Can it be a stack memory?

Reply:
I think the reason heap was used here is because there is a reasonable chance someone would run out of stack memory if we used stack (dynamic allocation of unknown very large size).

Reply:
How big do you think it will be?

Reply:
I don't know how you got that number but I don't think that is a right answer.

It is important to understand where the allocated memory comes from.
In this case, if you put `CapabilityDef` in a stack memory, it will occupy the stack memory as much as `sizeof(CapabilityDef)` tells us.
When I tried locally, I see "sizeof(CapabilityDef)" returns 176 (byte).

Note that the class has member variables that allocate from the heap memory.
When "CapabilityDef" is allocated from a stack memory, it doesn't make its member variables to allocate the memory from stack when they are calling `new` internally.

Reply:
I will just move it to the stack, no need to speculate on memory sizes I suppose.